### ISSUE TITLE
Bundling/bulk operations/event/something for more intensive events

### ISSUE THREAD

**Issue by xibz** [cdevents/spec#152]
Currently there are events that may produce a large number of subsequent events when they don't need to. For example, a recent deployment I tested against contained over 500 artifacts. That would be an event for each artifact.

This poses 2 problems:
1. The first being the amount of load this would create on the client/servers to send 500 events for a single deployment.
2. Things currently are very siloed, but certain events could be grouped together, e.g. test suites and tests cases, deployment and artifacts, etc...

Im still not sure if this is a spec issue or an engineering issue on the event bus/collector. However, the collector only really solves number 1. It could solve 2, but it could get a little more complicated there. So maybe a separate bundling event or something of that nature seems appropriate.

**Comment by e-backmark-ericsson** [comment]
I would like to know a bit more about that scenario where you had 500 "artifacts" in one deployment. It could come down to how we define the term "artifact". One artifact isn't necessarily equal to one binary, at least not in my mind. One artifact is an entity which has a specific identity and version. It could potentially be a collection of many binaries.
It's finally up to the producer of the artifact event to decide what types of (levels of) artifacts it wants to announce through the event. If there is a logical entity that has a unique identity and version, but includes 500 binaries, I think it is ok that just one artifact.packaged event is sent for the whole thingy. Unless the producer wants the event consumers to know about each individual 500 binaries.

**Comment by xibz** [comment]
> 500 "artifacts" in one deployment

This was one example, but based on your replies that opens up far more questions, rather than somehow supporting this. I think rather than opening up with the questions first, let me first ask this.

I mentioned also, test suites and tests cases, where there can be many test cases and you want to send that one by one rather than bulked?

**Comment by xibz** [comment]
> it's up to the producer to decide

Exactly. So why not allow for bulking if they want to send all test cases?

**Comment by e-backmark-ericsson** [comment]
Maybe it could be allowed.

We want the protocol to define one way, and one way only, to notify about an occurrence and its details. It makes it a lot harder for the consumers if the same information can be propagated in multiple ways and using multiple different event types.

Do you have a syntax proposal on how such a bundling could be done?
